## **Day 18 of 30 days Data Analysis** 

In [15]:
import pandas as pd

df = pd.read_csv("C:\\Users\\LENOVO\\Documents\\Programming\\30days_data_analysis\\week02_sql\\supermarket_d14.csv")

In [16]:
import sqlite3

conn = sqlite3.connect(":memory:")
df.to_sql("supermarket", conn, index=False, if_exists="replace")

1000

### ***Query1 - Average Rating by Product line:***

In [17]:
query1 = """
SELECT [Product line], AVG(Rating) as avg_rating
FROM supermarket
GROUP BY [Product line]
ORDER BY avg_rating DESC
"""

print(pd.read_sql_query(query1, conn))

             Product line  avg_rating
0      Food and beverages    7.113218
1     Fashion accessories    7.029213
2       Health and beauty    7.003289
3  Electronic accessories    6.924706
4       Sports and travel    6.916265
5      Home and lifestyle    6.837500


*- The product line with the highest average rating is `Food and beverages` with avg `7.11`.*

### ***Query2 - Top 5 Customers by Total Sales:***

In [18]:
query2 = """
SELECT [Invoice ID], SUM(Sales) as total_sales
FROM supermarket
GROUP BY [Invoice ID]
ORDER BY total_sales DESC
LIMIT 5
"""

print(pd.read_sql_query(query2, conn))

    Invoice ID  total_sales
0  860-79-0874      1042.65
1  687-47-8271      1039.29
2  283-26-5248      1034.46
3  751-41-9720      1023.75
4  303-96-2227      1022.49


*- The top customer had sales of `860-79-0874` with total sales `1042.65`.*

### ***Query3 - Sales trend by Month:***

In [19]:
query3 = """
SELECT [Month], SUM(Sales) as total_sales
FROM supermarket
GROUP BY [Month]
ORDER BY [Month]
"""

print(pd.read_sql_query(query3, conn))

   Month  total_sales
0      1   116291.868
1      2    97219.374
2      3   109455.507


*- Sales peaked in Month `January` with total sales `116291.86`.*

### ***Final Insight:***

*- SQL queries reveal key business patterns directly from the dataset.*

*- These insights can then be combined with Python visualizations for deeper analysis.*